### PR TITLE
making iterateRight an extension of Iterator 

### PR DESCRIPTION
**Pull request by plastic-karma**
Fix https://github.com/arrow-kt/arrow/issues/1428.

### CODE REVIEW

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Eval.kt [hunk 1]
```diff
@@ -271,3 +271,10 @@ sealed class Eval<out A> : EvalOf<A> {
     }
   }
 }
+
+fun <A, B> Iterator<A>.iterateRight(lb: Eval<B>): (f: (A, Eval<B>) -> Eval<B>) -> Eval<B> =
```
Comment (pakoito) [inline modules/core/arrow-core-data/src/main/kotlin/arrow/core/Eval.kt]: Almost there! This function returns another function, which isn't necessary in Kotlin. Make `f` a second parameter instead.

**Comment by pakoito** [comment]
For the build to pass you'll also need to fix:

```
/home/travis/build/arrow-kt/arrow/modules/core/arrow-extras-data/src/main/kotlin/arrow/data/MapK.kt:13:1: Unused import
/home/travis/build/arrow-kt/arrow/modules/core/arrow-extras-data/src/main/kotlin/arrow/data/SortedMapK.kt:11:1: Unused import
```

**Comment by plastic-karma** [comment]
> For the build to pass you'll also need to fix:
> 
> ```
> /home/travis/build/arrow-kt/arrow/modules/core/arrow-extras-data/src/main/kotlin/arrow/data/MapK.kt:13:1: Unused import
> /home/travis/build/arrow-kt/arrow/modules/core/arrow-extras-data/src/main/kotlin/arrow/data/SortedMapK.kt:11:1: Unused import
> ```

Great, I'll fix it. Is there a gradle command I could have run to have caught this before pushing ?

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Try.kt [hunk 2]
```diff
@@ -43,9 +43,6 @@ sealed class Try<out A> : TryOf<A> {
         }
       }
 
-    @Deprecated(DeprecatedAmbiguity, ReplaceWith("raiseError(e)"))
-    fun raise(e: Throwable): Try<Nothing> = Failure(e)
-
```
Comment (pakoito) [inline modules/core/arrow-core-data/src/main/kotlin/arrow/core/Try.kt]: These looks like a merge error to me! 😱

**Comment by pakoito** [comment]
>Great, I'll fix it. Is there a gradle command I could have run to have caught this before pushing ?

Yep, there's the build where you can check the tasks run. In this case, `:arrow-extras-data:ktlintMainSourceSetCheck`

https://travis-ci.org/arrow-kt/arrow/jobs/526942020#L5284-L5286

File: modules/core/arrow-core-data/src/main/kotlin/arrow/core/Try.kt (hunk 2 above)
Comment (plastic-karma) [reply modules/core/arrow-core-data/src/main/kotlin/arrow/core/Try.kt]: Yeah, I tried to rebase the branch to the latest version from the upstream fork

**Comment by plastic-karma** [comment]
I created a new request (after messing up the branch): https://github.com/arrow-kt/arrow/pull/1434

**Pull request closed without merging**